FirebaseRemoteConfig/Tests/Unit/RCNInstanceIDTest.m

```diff
@@ -0,0 +1,400 @@
+- (void)testNilInstanceIDWithValidInstanceIDToken {
+  NSMutableArray<XCTestExpectation *> *expectations =
+      [[NSMutableArray alloc] initWithCapacity:RCNTestRCNumTotalInstances];
+
+  // Test for each RC FIRApp, namespace instance.
+  for (int i = 0; i < RCNTestRCNumTotalInstances; i++) {
+    // Set the token as nil.
+    [self mockInstanceIDMethodForTokenAndIdentity:@"abcd"
+                                       tokenError:nil
+                                         identity:nil
+                                    identityError:nil];
+
+    expectations[i] =
+        [self expectationWithDescription:
+                  [NSString stringWithFormat:@"Test fetch configs successfully - instance %d", i]];
+    XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusNoFetchYet);
+    FIRRemoteConfigFetchCompletion fetchCompletion =
+        ^void(FIRRemoteConfigFetchStatus status, NSError *error) {
+          XCTAssertNotNil(error);
+          [expectations[i] fulfill];
+        };
+    [_configInstances[i] fetchWithExpirationDuration:43200 completionHandler:fetchCompletion];
+  }
+
+  [self waitForExpectationsWithTimeout:_expectationTimeout
+                               handler:^(NSError *error) {
+                                 XCTAssertNil(error);
+                               }];
+}
+
+// Instance ID is not nil, but IID SDK returns an error. Also token is valid.
+- (void)testValidInstanceIDAndError {
+  NSMutableArray<XCTestExpectation *> *expectations =
+      [[NSMutableArray alloc] initWithCapacity:RCNTestRCNumTotalInstances];
+
+  // Test for each RC FIRApp, namespace instance.
+  for (int i = 0; i < RCNTestRCNumTotalInstances; i++) {
+    // Set the token as nil.
+    [self mockInstanceIDMethodForTokenAndIdentity:@"abcd"
+                                       tokenError:nil
+                                         identity:@"test-id"
+                                    identityError:[NSError errorWithDomain:@"com.google.instanceid"
+                                                                      code:FIRInstanceIDErrorUnknown
+                                                                  userInfo:nil]];
+
+    expectations[i] =
+        [self expectationWithDescription:
+                  [NSString stringWithFormat:@"Test fetch configs successfully - instance %d", i]];
+    XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusNoFetchYet);
+    FIRRemoteConfigFetchCompletion fetchCompletion =
+        ^void(FIRRemoteConfigFetchStatus status, NSError *error) {
+          XCTAssertNotNil(error);
+          [expectations[i] fulfill];
+        };
+    [_configInstances[i] fetchWithExpirationDuration:43200 completionHandler:fetchCompletion];
+  }
+
+  [self waitForExpectationsWithTimeout:_expectationTimeout
+                               handler:^(NSError *error) {
+                                 XCTAssertNil(error);
+                               }];
+}
+
```

Comment:
Maybe add a test for a success case?

Reply:
This should be addressed in the other tests.